tools/ubuntu-setup/java8.sh

```diff
@@ -19,10 +19,20 @@
 set -e
 set -x
 
-if [ "$(lsb_release -cs)" == "trusty" ]; then
+JAVA_SOURCE=${1:-"open"}
+
+if [ "$JAVA_SOURCE" != "oracle" ] ; then
+    if [ "$(lsb_release -cs)" == "trusty" ]; then
+        sudo apt-get install -y software-properties-common python-software-properties
+        sudo add-apt-repository ppa:jonathonf/openjdk -y
```

Comment:
Not sure if this PPA build is "trusted"... https://askubuntu.com/questions/464755/how-to-install-openjdk-8-on-14-04-lts ... perhaps we should discuss this on mailing list?

Reply:
https://launchpad.net/~jonathonf/+archive/ubuntu/openjdk

Reply:
sudo add-apt-repository ppa:openjdk-r/ppa -y should be able to work.